File: modules/common/examples/widgets/org.eclipse.scout.widgets.client/src/main/java/org/eclipse/scout/widgets/client/services/lookup/ImageLookupCall.java
/*
 * Copyright (c) 2021 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Distribution License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/org/documents/edl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 */
package org.eclipse.scout.widgets.client.services.lookup;

import java.util.ArrayList;
import java.util.List;

import org.eclipse.scout.rt.platform.classid.ClassId;
import org.eclipse.scout.rt.shared.services.lookup.LocalLookupCall;
import org.eclipse.scout.rt.shared.services.lookup.LookupRow;

@ClassId("9d68e30d-9529-48f6-b6ca-707e45bef7a7")
public class ImageLookupCall extends LocalLookupCall<String> {

  private static final long serialVersionUID = 1L;

  @Override
  protected List<LookupRow<String>> execCreateLookupRows() {
    ArrayList<LookupRow<String>> rows = new ArrayList<>();
    rows.add(new LookupRow<>("application_logo.png", "application_logo.png"));
    rows.add(new LookupRow<>("fish.jpg", "fish.jpg"));
    return rows;
  }
}

File: code/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/image/ImageLookupCall.ts
/*
 * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 */
import {StaticLookupCall} from '@eclipse-scout/core';

export class ImageLookupCall extends StaticLookupCall<string> {

  constructor() {
    super();
  }

  protected override _data(): any[] {
    return ImageLookupCall.DATA;
  }

  static DATA = [
    ['img/eclipse_scout_logo.png', 'img/eclipse_scout_logo.png'],
    ['img/fish.jpg', 'img/fish.jpg'],
    ['img/bird.jpg', 'img/bird.jpg']
  ];
}
